io/disk/arcconf/arcconf_migration.py

```diff
@@ -0,0 +1,250 @@
+
+        # Gets the list of PCIIDs on the system
+        cmd = 'for device in $(lspci | awk \'{print $1}\') ; do echo \
+              $(lspci -vmm -nn -s $device | grep "\[" | awk \'{print $NF}\' \
+              | sed -e "s/\]//g" | sed -e "s/\[//g" | tr \'\\n\' \' \' \
+              | awk \'{print $2,$3,$4,$5}\') ; done'
+        pci_id_formatted = []
+        for i in self.cmdop_list(cmd).splitlines():
+            pci_id_formatted.append(str(i.replace(" ", ":")))
+
+        # check if all the yaml parameters are entered
+        if self.crtl_no is '' or self.channel_no is '' or self.disk_no is \
+           '' or self.pci_id is '' or len(self.disk_no) <= 1 or \
+           self.disk_size is '' or self.tool_name is '' or self.http_path is \
+           '' or self.initial_raid is '' or self.migrate_raid is '' or \
+           self.disk_initial is '' or self.disk_migrated is '' or \
+           len(self.initial_raid) > 1 or self.migration_sleep is '':
+            self.skip(" please ensure yaml parameters are not empty or \
+                       the total device should be more than 1")
+        elif self.comp(self.pci_id, pci_id_formatted) == 1:
+            self.skip(" Test skipped!!, PMC controller not available")
+
+        detected_distro = distro.detect()
+        if not smm.check_installed("arcconf"):
+            if detected_distro.name == "Ubuntu":
+                http_repo = "%s%s.deb" % (self.http_path, self.tool_name)
+                self.repo = self.fetch_asset(http_repo, expire='10d')
+                cmd = "dpkg -i %s" % self.repo
+            else:
+                http_repo = "%s%s.rpm" % (self.http_path, self.tool_name)
+                self.repo = self.fetch_asset(http_repo, expire='10d')
+                cmd = "rpm -ivh %s" % self.repo
+            if process.system(cmd, ignore_status=True, shell=True) == 0:
+                self.skip("Unable to install arcconf")
+
+        self.os_drive = ""
+        cmd = "lsscsi  | grep LogicalDrv | awk \'{print $7}\'"
+        self.os_drive = self.cmdop_list(cmd)
+
+        if self.diskoverwrite == 'Y' and self.os_drive != "":
+            # Delete a logical drive only if it is not OS drive
+            cmd = "df -h /boot | grep %s" % self.os_drive
+            if process.system(cmd, timeout=300, ignore_status=True,
+                              shell=True) == 0:
+                self.skip("Test Skipped!! OS disk requested for removal")
+
+            self.log.info("Deleting the default logical drive %s" %
+                          (self.logicaldrive))
+            cmd = "echo y | arcconf delete %s logicaldrive %s" % \
+                (self.crtl_no, self.logicaldrive)
+            self.check_pass(cmd, "Failed to delete Logical drive")
+
+        if len(self.disk_no) < int(self.disk_initial) or \
+           len(self.disk_no) < int(self.disk_migrated):
+            self.skip("Cannot Migrate, please check the prerequisite")
```

Comment:
done. this will be checked soon after the empty string check from yaml.


        if self.crtl_no is '' or self.channel_no is '' or self.disk_no is \
           '' or self.pci_id is '' or len(self.disk_no) <= 1 or \
           self.disk_size is '' or self.tool_name is '' or self.http_path is \
           '' or self.initial_raid is '' or self.migrate_raid is '' or \
           self.disk_initial is '' or self.disk_migrated is '' or \
           len(self.initial_raid) > 1 or self.migration_sleep is '':
            self.skip(" please ensure yaml parameters are not empty or \
                       the total device should be more than 1")
        elif self.comp(self.pci_id, pci_id_formatted) == 1:
            self.skip(" Test skipped!!, PMC controller not available")

        if len(self.disk_no) < int(self.disk_initial) or \
           len(self.disk_no) < int(self.disk_migrated):
            self.skip("Cannot Migrate, please check the prerequisite")